### DISCUSSION TITLE
How best to chunk a dynamic target into a new number of chunks?

### DISCUSSION THREAD

**Discussion by kendonB** [ropensci/targets#968] [Help]
I'm trying to save a large dynamic target of length N to disk in n chunks with n << N. There is not enough memory for n = 1.

My first idea was to get N programmatically (how do I do that? there is no `tar_length`) then chunk it up using `parallel::splitIndices`, read using `tar_read`, then save.

But there may be a better way.

I also wonder if this is a common enough workflow pattern to warrant a `pattern = chunk(group_target, n)` which would map through a fixed number of chunks but efficiently chunk by only reading in the current chunk's data from `group_target`.

**Comment by wlandau** [comment]
> My first idea was to get N programmatically (how do I do that? there is no tar_length) then chunk it up using parallel::splitIndices, read using tar_read, then save.

`length(tar_meta(names = dynamic_target)$children[[1]])` gives the number of branches of a dynamic target. Also things like `nrow(tar_branches(dynamic_target, pattern = map(mapped_over_target)))`.

> I also wonder if this is a common enough workflow pattern to warrant a pattern = chunk(group_target, n) which would map through a fixed number of chunks but efficiently chunk by only reading in the current chunk's data from group_target.

I am trying to handle this in `tarchtypes`. If the mapped-over target is a data frame, you can use [`tar_group_count()`](https://docs.ropensci.org/tarchetypes/reference/tar_group_count.html). Vectors and lists can easily be expressed as columns of data frames, so this should cover a lot in practice.
